lib/hooks.rb

```diff
@@ -1,17 +1,23 @@
 
 class Hooks
-  def self.run(kind, failover_ip, old_ip, new_ip)
+  def self.run(kind, failover_ip, old_ip, new_ip, dry = false)
     Dir[File.expand_path("../../hooks/#{kind}/*", __FILE__)].sort.each do |file|
-      system(file, failover_ip, old_ip, new_ip) if File.executable?(file)
+      if File.executable?(file)
```

Comment:
```suggestion
      next unless File.executable?(file)
```

Reply:
I can include your proposed change here. However I'd like to present my reasons for not doing so: as a general rule "unless" makes my brain trip over when reading code. Might be that me not being a Ruby coder I'm just not accustomed to the "unless" idiom. The problem is, "unless" is similar to a double negation: you only do something if some other thing is *not* true. That takes considerably more brain gymnastics to process than a simple if -> then. I'd argue that my coding maxime is "ease of understanding the code generally trumps everything".

In this specific case here not only we have this brain-flow-over-tripper, we in addition are directing control flow not via the property we are interested in towards the thing we are interested in (that is triggering an *executable* hook), but instead we're directing control flow based on the thing we're *not* interested in towards the thing we're not interested in.

So hmmn, I'm not excited by this change. Please reconfirm your preference and I'll implement it as you suggested.

Reply:
this is common in ruby, but i understand what you mean.
I'll change my suggestion then to already remove non-executables from the list:

```ruby
   files = Dir[File.expand_path("../hooks/#{kind}/*", __dir)]

   files.sort.select { |file| File.executable?(file) }.each do |file|
     # ...
  end
```

Reply:
updated my comment as it should be `__dir__`, not `__dir`

Reply:
very nice! Thanks -> 1d1ddca